core/src/test/java/io/questdb/test/griffin/O3ParquetPartitionFuzzTest.java

```diff
@@ -0,0 +1,202 @@
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin;
+
+import io.questdb.cairo.*;
+import io.questdb.cairo.sql.RecordCursor;
+import io.questdb.cairo.sql.RecordCursorFactory;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.cairo.sql.TableMetadata;
+import io.questdb.griffin.SqlCompiler;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.std.ObjList;
+import io.questdb.std.Rnd;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.StringSink;
+import io.questdb.test.fuzz.FuzzTransaction;
+import io.questdb.test.fuzz.FuzzTransactionGenerator;
+import io.questdb.test.fuzz.FuzzTransactionOperation;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Test;
+
+public class O3ParquetPartitionFuzzTest extends AbstractO3Test {
+
+    public static GenericRecordMetadata symbolAsVarcharCopy(RecordMetadata that) {
+        if (that != null) {
+            GenericRecordMetadata metadata = new GenericRecordMetadata();
+            for (int i = 0, n = that.getColumnCount(); i < n; i++) {
+                final int columnType = that.getColumnType(i);
+                metadata.add(
+                        new TableColumnMetadata(
+                                that.getColumnName(i),
+                                columnType == ColumnType.SYMBOL ? ColumnType.VARCHAR : columnType,
+                                that.isColumnIndexed(i),
+                                that.getIndexValueBlockCapacity(i),
+                                that.isSymbolTableStatic(i),
+                                that.getMetadata(i),
+                                that.getWriterIndex(i),
+                                that.isDedupKey(i)
+                        )
+                );
+            }
+            metadata.setTimestampIndex(that.getTimestampIndex());
+            return metadata;
+        }
+        return null;
+    }
+
+    @Test
+    public void testFuzz() throws Exception {
+        executeWithPool(0, this::testFuzz0);
```

Comment:
This test generates parquet partitions with only 1 row group, please add tests that generate parquet partitions with many row groups.